lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,1936 @@
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating tagging parameters - " + str(e) +
+                "\n" + traceback.format_exc())
+
+    def validate_tags(self, tags):
+        try:
+            if tags is None:
+                return None
+            list_items = []
+            for tag in tags:
+                key = tag.keys()[0]
+                value = str(tag[key])
+                list_items.append({
+                    'Key': key,
+                    'Value': value
+                })
+            return list_items
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating tags - " + str(e) +
+                "\n" + traceback.format_exc())
+
+    def validate_distribution_config_parameters(
+            self, config, default_root_object, is_ipv6_enabled, http_version,
+            web_acl_id):
+        try:
+            config['default_root_object'] = (default_root_object or '')
+            config['is_i_p_v_6_enabled'] = (is_ipv6_enabled or
+                                            self.__default_is_ipv6_enabled)
+            if http_version is not None:
+                self.validate_attribute_with_allowed_values(
+                    http_version, 'http_version', self.__valid_http_versions)
+                config['http_version'] = http_version
+            if web_acl_id is not None:
+                config['web_a_c_l_id'] = web_acl_id
+            return config
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating distribution config parameters - " +
+                str(e) + "\n" + traceback.format_exc())
+
+    def validate_streaming_distribution_config_parameters(
+            self, config, comment, trusted_signers, s3_origin,
+            default_s3_origin_domain_name,
+            default_s3_origin_origin_access_identity):
+        try:
+            if s3_origin is None:
+                s3_origin = config.get('s3_origin')
+            config['s3_origin'] = self.validate_s3_origin(
+                s3_origin, default_s3_origin_domain_name,
+                default_s3_origin_origin_access_identity)
+            config['trusted_signers'] = self.validate_trusted_signers(
+                trusted_signers)
+            return config
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating streaming distribution config " +
+                "parameters - " + str(e) + "\n" + traceback.format_exc())
+
+    def validate_common_distribution_parameters(self, config, enabled, aliases,
+                                                logging, price_class, comment,
+                                                is_streaming_distribution):
+        try:
+            if config is None:
+                config = {}
+            if aliases is not None:
+                config['aliases'] = helpers.python_list_to_aws_list(aliases)
+  
```

Comment:
Thanks for the feedback @s-hertel ! Yes, that does seem unexpected. However, I'm unable to run that as a task as either a distribution_id, caller_reference or an alias is usually required for updating a distribution and either default_origin_domain_name or origins[] are required for a new distribution. Are there extra parameters you are running this task with? The playbook I'm using is:

```
- hosts: 127.0.0.1
  connection: local
  tasks:
    - cloudfront_distribution:
        state: present
        enabled: false
        e_tag: "E1I5YC9H172ZQX"
```

(after fixing a minor bug) the output I'm getting is:

` "msg": "both origins[] and default_origin_domain_name have not been specified. please specify at least one."`